Update Installation Troubleshooting Documentation (#4007)
ros2/ros2_documentation: source/How-To-Guides/Installation-Troubleshooting.rst

   colcon version-check  # check if newer versions available
   sudo apt install python3-colcon* --only-upgrade  # upgrade installed colcon packages to latest version

Anaconda Python Conflict
^^^^^^^^^^^^^^^^^^^^^^^^

``conda`` does not work in conjunction with ROS 2.
Make sure that your ``PATH`` environment variable does not have any conda paths in it.

> You may have to check your `.bashrc` for this line and comment it out.

Reviewer: ```suggestion
You may have to check your ``.bashrc`` for this line and comment it out.
```

> 

Reviewer: Sorry, I just noticed that this is under the "Linux" heading above.  Because conda is cross-platform, I think this should go under the "General" heading even further above.

Author: I don't know if this happens on anything other than Linux though...

Reviewer: > I don't know if this happens on anything other than Linux though...

I'm fairly certain it will, but that's a fair point.  We can leave it under the Linux heading for now.

   colcon version-check  # check if newer versions available
   sudo apt install python3-colcon* --only-upgrade  # upgrade installed colcon packages to latest version

Anaconda Python Conflict
^^^^^^^^^^^^^^^^^^^^^^^^

> ``conda`` does not play nice with ROS2. To fix this make sure your ``PATH`` environment variable does not have any conda paths in it. You may have to check your `.bashrc` for this line and comment it out.

Reviewer: ```suggestion
``conda`` does not work in conjunction with ROS 2.
Make sure that your ``PATH`` environment variable does not have any conda paths in it.
You may have to check your `.bashrc` for this line and comment it out.
```

> .. code-block:: bash
>    export PATH="/home/foo/miniconda3/bin:$PATH"
> 

Reviewer: I'm not sure we need this, as it is as confusing PATH that will be different for everyone.  I think the advice above is good enough, so:

```suggestion
```
